lib/ood_core/job/adapters/slurm.rb

```diff
@@ -98,6 +98,23 @@ def initialize(cluster: nil, bin: nil, conf: nil, bin_overrides: {}, submit_host
             @strict_host_checking = strict_host_checking
           end
 
+          # Get a ClusterInfo object containing information about the given cluster
+          # @return [ClusterInfo] object containing cluster details
+          def get_cluster_info
+            sinfo_out = call("sinfo", "-aho %A/%D/%C").strip.split('/')
+            sinfo_out2 = call("sinfo", "-Nhao %n/%G/%T").lines.uniq
+            gpu_total = sinfo_out2.grep(/gpu:/).count
+            gpu_free = sinfo_out2.grep(/gpu:/).grep(/idle/).count
```

Comment:
`sinfo -Nhao %n/%G/%T` I think this is just node status. I don't know if `allocated` means the GRES was also allocated. Meaning, I have a job on a node that has a GPU, but I didn't request or allocate it.


I think we actually want something more like `sinfo -Nha -O 'GresUsed'` that indicates whether the GRES is in use or not (as a pose to the the Node being in use or not).

  @treydock please advise -

Reply:
Oh i see, we're returning `gpu_nodes_active`. No I think the better information is GPUs active. There are multiple GPUs on a given node.

I think reporting # of GPU Nodes is secondary to reporting on # of GPUs themselves.

Reply:
Collecting GRES used with sinfo is actually a complicated process.  I ran into this with Slurm exporter.  The information about available/used GRES is only available via `--Format` flag so the printf `%G` isn't possible to get used, that's just available.

First you must get longest GRES line (they can get very long). Also you can likely cache this as GRES definitions only change when slurm.conf and gres.conf are changed and slurmctld is restarted.

```
sinfo -o '%G'
```

Then get GRES data:

```
sinfo -a -h --Node --Format=nodehost,gres:%d,gresused%d
```

You replace `%d` with value of longest GRES plus a few if you want a buffer.

Reply:
In case not clear, the `sinfo -o '%G'` you have to iterate over all lines to find longest and use that for the length with `--Format`